src/decaylanguage/dec/dec.py

```diff
@@ -281,18 +294,80 @@ def load_grammar(
         for parser, lexer and options.
         """
 
+        warnings.warn(
+            "This method is deprecated, please add unknown decay models by passing them to "
+            "`load_additional_decay_models` instead before parsing the decayfile.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         if filename is None:
             filename = "decfile.lark"
-            with data.basepath.joinpath(filename).open() as f1:
-                self._grammar = f1.read()
+
+        self._load_grammar(filename, parser, lexer, **options)
+
+    def load_additional_decay_models(self, *models: str) -> None:
+        if self._additional_decay_models is None:
+            self._additional_decay_models = known_decay_models
         else:
-            with Path(filename).open(encoding="utf_8") as f2:
-                self._grammar = f2.read()
+            self._additional_decay_models = chain.from_iterable(
+                (self._additional_decay_models, models)
+            )
+
+    def _load_grammar(
+        self,
+        filename: str = "decfile.lark",
+        parser: str = "lalr",
+        lexer: str = "auto",
+        **options: Any,
+    ) -> None:
+        """
+        Load the default Lark grammar definition file with default parser and lexer.
+
+        Parameters
+        ----------
+        filename: str, optional, default='decfile.lark'
+            The Lark grammar file name.
+        parser: str, optional, default='lalr'
+            The Lark parser engine name.
+        lexer: str, optional, default='auto'
+            The Lark parser lexer mode to use.
+        options: keyword arguments, optional
+            Extra options to pass on to the parsing algorithm.
+        """
+
+        with data.basepath.joinpath(filename).open() as f1:
+            self._grammar = f1.read()
 
         self._grammar_info = dict(
-            lark_file=filename, parser=parser, lexer=lexer, **options
+            lark_file=filename,
+            parser=parser,
+            lexer=lexer,
+            edit_terminals=self._generate_edit_terminals_callback(),
+            **options,
         )
 
+    def _generate_edit_terminals_callback(self) -> Callable[[TerminalDef], None]:
```

Comment:
Can you also add some docstring here motivating this internal helper? It does help when one reads things back months down the line :-).